src/main/java/com/minecolonies/coremod/colony/Colony.java

```diff
@@ -1208,55 +853,13 @@ private boolean areAllColonyChunksLoaded(@NotNull final TickEvent.WorldTickEvent
         return true;
     }
 
-    private void cleanUpBuildings(@NotNull final TickEvent.WorldTickEvent event)
-    {
-        @Nullable final List<AbstractBuilding> removedBuildings = new ArrayList<>();
-
-        //Need this list, we may enter he while we add a building in the real world.
-        final List<AbstractBuilding> tempBuildings = new ArrayList<>(buildings.values());
-
-        for (@NotNull final AbstractBuilding building : tempBuildings)
-        {
-            final BlockPos loc = building.getLocation();
-            if (event.world.isBlockLoaded(loc) && !building.isMatchingBlock(event.world.getBlockState(loc).getBlock()))
-            {
-                //  Sanity cleanup
-                removedBuildings.add(building);
-            }
-        }
-
-        @NotNull final ArrayList<BlockPos> tempFields = new ArrayList<>(fields);
-
-        for (@NotNull final BlockPos pos : tempFields)
-        {
-            if (event.world.isBlockLoaded(pos))
-            {
-                final ScarecrowTileEntity scarecrow = (ScarecrowTileEntity) event.world.getTileEntity(pos);
-                if (scarecrow == null)
-                {
-                    fields.remove(pos);
-                }
-            }
-        }
-
-        removedBuildings.forEach(AbstractBuilding::destroy);
-    }
-
-    /**
-     * Spawn a brand new Citizen.
-     */
-    private void spawnCitizen()
-    {
-        spawnCitizen(null);
-    }
-
     private void updateOverallHappiness()
     {
         int guards = 1;
         int housing = 0;
         int workers = 1;
         double saturation = 0;
-        for (final CitizenData citizen : citizens.values())
+        for (final CitizenData citizen : citizenManager.getCitizens())
```

Comment:
This should probably be moved into the citizenmanager

Reply:
Yeah, good idea.